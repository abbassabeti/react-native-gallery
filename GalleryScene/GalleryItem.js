// Imports: Dependencies
import React from 'react';
import {Image } from 'react-native';
import {Dimensions, TouchableOpacity, StyleSheet, Text } from 'react-native';
//import CacheableImage from 'react-native-cacheable-image'

// Screen Dimensions
const { height, width } = Dimensions.get('window');
// Screen: Infinite Scroll

export default GalleryItem = (id,title,navigateTo) => {
  let imgId = id && id.id ? id.id : id
  let imgTitle = id && id.title ? id.title : title
  let navFunc = id && id.navigateTo ? id.navigateTo : navigateTo
  return (
      <TouchableOpacity
        testID={`touchable${imgId}`}
        style={styles.item}
        onPress={()=> navFunc("image",{imageId:imgId,imgTitle})}>
        {/*<CacheableImage
            style={styles.image}
            source={{uri:`https://picsum.photos/id/${id}/200`}}
            checkNetwork
            networkAvailable
        >
        </CacheableImage>*/}
          <Image
            style={styles.image}
            id={imgId}
          source={{uri:`https://picsum.photos/id/${imgId}/200`}}>
          </Image>
      </TouchableOpacity>
    )};

  // Styles
const styles = StyleSheet.create({
  item: {
    backgroundColor: '#cccccc',
    marginVertical: 8,
    marginHorizontal: 16,
    flex: 1,
    aspectRatio: 1,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'transparent',
    overflow: 'hidden'
  },
  image:{
    width: '100%',
    height: '100%',
  },
});