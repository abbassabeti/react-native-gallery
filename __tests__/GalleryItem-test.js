import React from 'react';
import { fireEvent, render } from '@testing-library/react-native'
import GalleryItem from '../GalleryScene/GalleryItem';
import { exp } from 'react-native-reanimated';


describe('Testing GalleryItem', () => {
    it(`GalleryItem test failed in snapshot or finding element`, () => {
      let k = 0
      let method = jest.fn()
      const componentTree = render(<GalleryItem id={4} title={'test'} navigateTo={/*()=>{k++}*/method}/>)
      const {getByTestId} = componentTree;
      const element = getByTestId(`touchable${4}`)
      fireEvent.press(element)
      expect(method).toBeCalled()
      expect(componentTree).toMatchSnapshot()
    })
})