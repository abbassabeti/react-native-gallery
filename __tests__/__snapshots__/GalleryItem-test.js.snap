// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FlatList Error component should be render when error is true 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "backgroundColor": "#cccccc",
      "flex": 1,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "opacity": 1,
    }
  }
  testID="touchable4"
>
  <Image
    id={4}
    source={
      Object {
        "uri": "https://picsum.photos/id/4/200",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
